arch/arc/include/swap_macros.h

```diff
@@ -86,9 +89,14 @@
 	lr r13, [_ARC_V2_FPU_DPFP2H]
 	st_s r13, [sp, ___callee_saved_stack_t_dpfp2h_OFFSET]
 #endif
-1 :
 #endif
-
+fpu_skip_save :
+#ifdef CONFIG_ARC_DSP_SHARING
```

Comment:
We do check against CONFIG_ARC_DSP_SHARING in the `_save_dsp_regs` macro itself. So let's keep the check only in the macro and drop it from here - so we'd get a bit cleaner code.

Reply:
done